### DISCUSSION TITLE
Emails don't seem to give expected results

### DISCUSSION THREAD

**Discussion by Komplex05** [dani-garcia/vaultwarden#2087] [Q&A] [answered]
Hey guys, 
I'm fairly new to home servers and I am running vaultwarden on unraid. I'm having a few issues and I've read over the instructions, searched the forums/google and I cant seem to find anyone that has my exact issue. When I attempt to send an email for registration when the user clicks on the email to take them to the VW page but when they click and try to register with the same email it states that the account already exist I bypassed this by deleting the users and letting them register. Clicking on verification emails or ORG emails don't seem to work either. The link takes them to the sign-in page but after signing in nothing changes.

**Comment by BlackDex** [comment]
Looks like the token verification is failing, or that the token is modified at the receivers end by the mail client or something.

What do the logs say during the time the user clicked this links and tried to register or verify?

**Comment by Komplex05** [comment]
I saw that in the logs as well but nothing I try seems to fix it. this is the log for verify.

[2021-11-04 19:39:00.367][vaultwarden::api::identity][ERROR] Please verify your email before trying again.. IP: 192.168.1.1. Username: dtest@thescholarsloft.org.

[2021-11-04 19:39:00.367][response][INFO] POST /identity/connect/token (login) => 400 Bad Request
[2021-11-04 19:40:33.567][request][INFO] POST /api/accounts/prelogin
[2021-11-04 19:40:33.571][response][INFO] POST /api/accounts/prelogin (prelogin) => 200 OK
[2021-11-04 19:40:33.628][request][INFO] POST /identity/connect/token

Here is the invite
[2021-11-07 07:35:12.894][request][INFO] POST /admin/invite/
[2021-11-07 07:35:14.366][response][INFO] POST /admin/invite (invite_user) => 200 OK
[2021-11-07 07:35:15.975][request][INFO] GET /admin/users/overview
[2021-11-07 07:35:15.976][response][INFO] GET /admin/users/overview (users_overview) => 200 OK
[2021-11-07 07:35:51.469][request][INFO] POST /api/accounts/register
[2021-11-07 07:35:51.470][vaultwarden::api::core::accounts][ERROR] Account with this email already exists

[2021-11-07 07:35:51.470][response][INFO] POST /api/accounts/register (register) => 400 Bad Request

**Reply by BlackDex** [reply]
The message `Account with this email already exists` only appears if the user does not use the link which holds the token while there was an e-mail send to the user which holds this. The same goes for the email verify, that needs a token, and if that token is not correct or not passed then it will fail.

**Reply by Komplex05** [reply]
So When I test it I am going straight from the inbox and clicking the link. Could there be something in the admin console that could be causing the issue?

**Reply by BlackDex** [reply]
But does that link have a token?
Without a token it will not work.

**Reply by Komplex05** [reply]
Is that something that needs to be changed in the config file? I tried looking for token creation but I wasn't able to find anything

**Reply by BlackDex** [reply]
No, that is by default. Invites always have a token in the link.
An URL should look something like this. (I have shortened the link and replaced some part with `......`)
`
https://vw.domain.tld/#/accept-organization/?organizationId=_&organizationUserId=_&email=invite%40vw%2Edomain%2Etld&organizationName=Vaultwarden&token=eyJ0eXAiOiJKV1QiLCJhbGc.....poHkwu2cqSq63FSEuw
`

**Reply by Komplex05** [reply]
Ok I do have a token...Its very very long.

**Reply by Komplex05** [reply]
I copied and pasted the link directly in my browser just to be sure and I still couldn't create an account from the invite

**Comment by Komplex05** [comment, marked as answer]
OK guys so I don't know if anyone will ever have this issue. I an using ngnix proxy manager for my reverse proxy. When I go to bitwarden.mydomain.com it redirects me to bitwarden.mydomain.com/#/ so when I went into my vaultwarden admin console to enter the Domain URL I naturally copied it from the browser and pasted it in the field. Apparently the /#/ caused vaultwarden to send a join link with https://bitwarden.mydoman.com/#//%23/accept-organization/ which caused nginx to redirect me to the "correct" page stripping the token from the url and sending me to the main page. After changing the domain URL to bitwarden.mydomain.com the invite was sent in the correct format https://bitwarden.mydoman.com/#/accept-organization/. Bottom line Is to make sure that your domain URL is correct and don't add the /#/ because it will cause major issues. @BlackDex Thank you for pointing me in the right direction. Your time and patience is appreciated. If there was something that I didn't explain correctly please correct me.
